//战斗相关
package words;
option py_generic_services=true;

import "universal/public.proto";
import "common.proto";

//内容
message content{
	optional int32 event=1;//事件 1:登场,2:出招,3:挨打,4:保护,5:主人倒地,6:使用道具,7:逃跑失败,8:主人倒地
	optional bytes content=2;//闲话内容
}

//闲话
message wordsInfo{
	required int32 type=1;//类型 1：异兽，2：伙伴
	required int64 entity=2;//伙伴编号或者宠物编号
	optional content msg=3;//闲话内容
}

//闲话列表
message wordsList{
	required int32 type=1;//类型 1：异兽，2：伙伴
	required int64 entity=2;//伙伴编号或者宠物编号
	repeated content msgList=3;//闲话内容
}

//上行(客户端主动发起)
service terminal2main{
	rpc rpcWordsGet(words.wordsInfo)returns(public.fake);//获取闲话
	rpc rpcWordsSet(words.wordsInfo)returns(public.fake);//设置闲话
	rpc rpcWordsReset(words.wordsInfo)returns(public.fake);//重置闲话，上传 类型和 编号
}

//下行(服务端主动发起)
service main2terminal{
	rpc rpcWordsList(words.wordsList)returns(public.fake);//闲话列表
	rpc rpcWordsMod(words.wordsInfo)returns(public.fake);//闲话修改
 }